src/frontend/Typechecker.ml

```diff
@@ -603,9 +603,23 @@ and check_expression cf tenv ({emeta; expr} : Ast.untyped_expression) :
                  operator %/%."
             in
             add_warning x.emeta.loc s
+        | UMatrix, (UInt | UReal), Pow ->
```

Comment:
I'd at least add `UArray UMatrix` here. Emitting a good warning in every case is hard because `^` is fully vectorized so that even `matrix ^ matrix` compiles.

Reply:
I don't believe `matrix_power()` is overloaded for arrays of matrices (if it is, it isn't documented as such). So the suggestion wouldn't be 100% a drop-in replacement in that case.

Reply:
It's not available for `UReal` exponent either. The warning still guards against misinterpretation.

Reply:
Good point. I'll extend the case, and specifically say "function matrix_power(matrix,int)" for maximum clarity in the hint.